src/core_scan/Core_scan.ml

```diff
@@ -916,19 +907,32 @@ let scan ?match_hook config ((rules, invalid_rules), rules_parse_time) :
            update_cli_progress config;
 
            (* adjust the match location for extracted targets *)
-           match Hashtbl.find_opt extract_result_map !!file with
-           | Some f -> f matches
-           | None -> matches)
+           let matches =
+             match Hashtbl.find_opt extract_result_map !!file with
+             | Some f -> f matches
+             | None -> matches
+           in
+           (matches, was_scanned))
+  in
+  let scanned_target_table =
+    (* provide fast access to paths that were scanned by at least one rule;
+       includes extracted targets *)
+    scanned_targets
+    |> Common.map (fun path -> (!!path, ()))
```

Comment:
Could perhaps use `Seq` here to avoid constructing a potentially huge auxiliary list? In monorepos this list is going to be very long, probably fills the minor heap, ends up being promoted to the major heap, and after `hash_of_list` it becomes unreachable and it needs to be GCed.

Reply:
I usually write `let tbl = Hashtbl.create (List.length list) in List.iter (fun x -> Hashtbl.replace tbl x ()) list;` because then I don't have to worry about performance being suboptimal. Note that the hash table itself does a fair amount of allocation ([one block per element like for a list](https://github.com/lucasaiu/ocaml/blob/323378a510bda2abd463cfa1888deb9a965e9e87/stdlib/hashtbl.ml#L35-L37)).

Reply:
I might add a utility function that creates the table in one shot:
```ocaml
val hash_of_keys : 'a list -> ('a, unit) Hashtbl.t 
```

Reply:
It turns out it already exists as `Common.hashset_of_list`.

Reply:
no it doesn't.